### PR TITLE
[CHK-2470] session token into outcome URL

### PR DESCRIPTION
**Pull request by Fernando-Granato**
#### List of Changes
Added jwt token for outcome page useful for fe to poll downstream of an APM payment with wallet.
Modified the logic to generate the 'outcome url.

#### Motivation and Context
The change was made to allow successful completion of an APM payment with the wallet.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java [hunk 1]
```diff
@@ -672,4 +665,61 @@ private Either<CheckoutRedirectConfigurationException, URI> getRedirectUrlForPsp
...
+                                                                        outcomeToken
+                                                                )
+                                                        )
+                                                )
+                                )
+
+                );
+    }
+
+    private URI generateNpgOutcomeUrl(
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:707]: 🚲  this method is used by both npg and redirect payment flows. What about refactoring it to 
```suggestion
    private URI generateOutcomeUrl(
```

Comment (Fernando-Granato) [reply src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java]: Fixed with c294728dd00f565352271c99e453544d4a954731

File: src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java [hunk 2]
```diff
@@ -543,107 +536,107 @@ public Mono<RedirectUrlResponseDto> requestRedirectUrlAuthorization(
...
-                                                "Unhandled error performing redirection URL api call for PSP: [%s]"
-                                                        .formatted(pspId),
-                                                null
-                                        );
-                                    }
-                                }
+        return new JwtTokenUtils()
+                .generateToken(
+                        npgNotificationSigningKey,
+                        npgJwtKeyValidityTime,
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java:542]:
```suggestion
                        npgNotificationSigningKey,
                        npgJwtKeyValidityTime,
```
This token is the one that will be used by eCommerce f.e. for polling the transaction status in the outcome page. Please use the eCommerce signing key here

Comment (Fernando-Granato) [reply src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java]: Excellent observation fixed with 56689e716478674f1d3421ef9eb15b8ebe407ecc

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**